solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -552,9 +557,9 @@ pub struct SendTransfer<'info> {
     mint_authority: Option<UncheckedAccount<'info>>,
     #[account(mut)]
     token_mint: Option<Box<Account<'info, Mint>>>,
-    // Splitting `base_denom` since each seed can be at most 32 byte long.
+    // Splitting `hashed_base_denom` since each seed can be at most 32 byte long.
     #[account(init_if_needed, payer = sender, seeds = [
-        port_id.as_bytes(), channel_id.as_bytes(), base_denom[..32].as_bytes(), base_denom[32..].as_bytes()
+        port_id.as_bytes(), channel_id.as_bytes(), hashed_base_denom[..32].as_ref(), hashed_base_denom[32..].as_ref()
```

Comment:
Hash is always 32 bytes so no need for splitting
```suggestion
        port_id.as_bytes(), channel_id.as_bytes(), hashed_base_denom.as_bytes()
```